### ISSUE TITLE
Installing Python3.10 on Jenkins Agent

### ISSUE THREAD

**Issue by vsudhire** [jenkinsci/docker-inbound-agent#353]
### Jenkins and plugins versions report

I am downloading latest Jenkins agent container image and when I am unable to install Python3.10. 

Here is my docker file content for example: 

FROM jenkins/inbound-agent:latest
# python packages
RUN apt-get update && apt upgrade -y \
  && apt install software-properties-common -y \
  && add-apt-repository ppa:deadsnakes/ppa \
  && apt-get install -y curl wget \
  && apt-get install -y python3.10 python3-pip python3-venv python3-dev

Statement "add-apt-repository ppa:deadsnakes/ppa" is reporting following warning 
For nightly builds, see ppa:deadsnakes/nightly https://launchpad.net/~deadsnakes/+archive/ubuntu/nightly
 More info: https://launchpad.net/~deadsnakes/+archive/ubuntu/ppa
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python3.9/threading.py", line 954, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.9/threading.py", line 892, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3/dist-packages/softwareproperties/SoftwareProperties.py", line 688, in addkey_func
    func(**kwargs)
  File "/usr/lib/python3/dist-packages/softwareproperties/ppa.py", line 386, in add_key
    return apsk.add_ppa_signing_key()
  File "/usr/lib/python3/dist-packages/softwareproperties/ppa.py", line 251, in add_ppa_signing_key
    if not self._recv_key(
  File "/usr/lib/python3/dist-packages/softwareproperties/ppa.py", line 176, in _recv_key
    res = subprocess.call(self.GPG_DEFAULT_OPTIONS + [
  File "/usr/lib/python3.9/subprocess.py", line 349, in call
    with Popen(*popenargs, **kwargs) as p:
  File "/usr/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.9/subprocess.py", line 1823, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'gpg'

--
When I tried to run python3.10 install command it's not working. i.e. it's not found as one of the package. 

Wanted to check if this combination has been tested & if there is anything that I can do to install python3.10, obviously trying to avoid downloading/cloning the repo and building locally on the agent itself. 

Thanks, 

### What Operating System are you using (both controller, and any agents involved in the problem)?

Linux

### Reproduction steps

Build Agent using below content in the Dockerfile. 

FROM jenkins/inbound-agent:latest
# python packages
RUN apt-get update && apt upgrade -y \
  && apt install software-properties-common -y \
  && add-apt-repository ppa:deadsnakes/ppa \
  && apt-get install -y curl wget \
  && apt-get install -y python3.10 python3-pip python3-venv python3-dev

### Expected Results

Python3.10 gets installed.

### Actual Results

Python3.10 didn't get installed as expected.

**Comment by timja** [comment]
Your error is `No such file or directory: 'gpg'`

i.e. missing the `gpg` command.

This isn't the appropriate place to ask for generic docker support.
Try on stackoverflow.com or docker forums

**Comment by gounthar** [comment]
What I can see in the logs of the ppa you added is:

> - Note: Python2.7 (all), Python 3.6 (bionic), Python 3.8 (focal), and Python 3.10 (jammy) are not provided by deadsnakes as upstream Ubuntu provides those packages.

So no wonder it was not found in this repo. 

I see you're starting from [`latest`](https://github.com/jenkinsci/docker-inbound-agent/blob/master/docker-bake.hcl#L94), which happens to be `jdk11` on top of `Debian`.
There is a [dependency](https://github.com/jenkinsci/docker-inbound-agent/blob/master/debian/Dockerfile#L3) to the base Jenkins agent image. This one is (as of the time I'm writing) based on [Debian Bullseye 20230502](https://github.com/jenkinsci/docker-agent/blob/master/debian/Dockerfile#L53).

So I ran this command to start the same base image as a container:

` docker run -it debian:bullseye-20230502 /bin/bash`

And I ran your commands to try and install Python...
It also failed when I tried to add your specific ppa. 🤷 

Installing directly Python `3.10` without your ppa does work though...

` apt-get install -y python3.10 python3-pip python3-venv python3-dev`


Anyway, I'm afraid this has nothing to do with Jenkins, it's just a misconfiguration on your side...


Oh, and while I was writing this message, Tim got the short version of my answer. 🤣
